libraries/botbuilder-azure/botbuilder/azure/blob_storage.py

```diff
@@ -23,56 +44,107 @@ def __init__(
             )
         else:
-            raise Exception(
-                "Connection string should be provided if there are no account name and key"
+            blob_service_client = BlobServiceClient.from_connection_string(
+                convert_account_name_and_key_to_connection_string(settings)
             )
 
-        self.client = client
-        self.settings = settings
+        self.__container_client = blob_service_client.get_container_client(
+            settings.container_name
+        )
+
+        self.__initialized = False
+
+    async def _initialize(self):
+        if self.__initialized is False:
+            # This should only happen once - assuming this is a singleton.
+            # ContainerClient.exists() method is available in an unreleased version of the SDK. Until then, we use:
+            try:
+                await self.__container_client.create_container()
+            except ResourceExistsError:
+                pass
+            self.__initialized = True
+        return self.__initialized
 
     async def read(self, keys: List[str]) -> Dict[str, object]:
+        """Retrieve entities from the configured blob container.
+
+        :param keys: An array of entity keys.
+        :type keys: Dict[str, object]
+        :return dict:
+        """
         if not keys:
             raise Exception("Keys are required when reading")
 
-        self.client.create_container(self.settings.container_name)
-        self.client.set_container_acl(
-            self.settings.container_name, public_access=PublicAccess.Container
-        )
+        await self._initialize()
+
         items = {}
 
         for key in keys:
-            if self.client.exists(
-                container_name=self.settings.container_name, blob_name=key
-            ):
-                items[key] = self._blob_to_store_item(
-                    self.client.get_blob_to_text(
-                        container_name=self.settings.container_name, blob_name=key
-                    )
-                )
+            blob_name = self._get_blob_name(key)
+            blob_client = self.__container_client.get_blob_client(blob_name)
+
+            try:
+                items[key] = await self._inner_read_blob(blob_client)
+            except HttpResponseError as err:
+                if err.status_code == 404:
+                    continue
 
         return items
 
     async def write(self, changes: Dict[str, object]):
+        """Stores a new entity in the configured blob container.
+
+        :param changes: The changes to write to storage.
+        :type changes: Dict[str, object]
+        :return:
+        """
         if changes is None:
             raise Exception("Changes are required when writing")
         if not changes:
             return
 
-        self.client.create_container(self.settings.container_name)
-        self.client.set_container_acl(
-            self.settings.container_name, public_access=PublicAccess.Container
```

Comment:
I didn't see this referenced in .NET or JS and tests all passed fine without it. I can add it back, if necessary.